### PR TITLE
Integrate llvm-project at f66cd9e9556a53142a26a5c21a72e21f1579217c

### PR DESCRIPTION
**Pull request by stellaraccident**
* Reset third_party/llvm-project: f66cd9e9556a53142a26a5c21a72e21f1579217c (2023-09-14 18:45:29 -0700): [mlir] Add Python bindings for DenseResourceElementsAttr. (#66319)

### CODE REVIEW

**Comment by Groverkss** [comment]
I did mostly mechanical changes:

```
find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/CodeGenOpt::Aggressive/CodeGenOptLevel::Aggressive/g"
find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/CGFT_/CodeGenFileType::/g"

find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/arith::MinF/arith::MinimumF/g"
find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/arith::MaxF/arith::MaximumF/g"

find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/TileToScfForOp/TileOp/g"
```

**Comment by MaheshRavishankar** [comment]
> I did mostly mechanical changes:
> 
> ```
> find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/CodeGenOpt::Aggressive/CodeGenOptLevel::Aggressive/g"
> find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/CGFT_/CodeGenFileType::/g"
> 
> find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/arith::MinF/arith::MinimumF/g"
> find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/arith::MaxF/arith::MaximumF/g"
> 
> find . -type f -name "*.cpp" -print0 -or -name "*.h" -print0 | xargs -0 sed -i "s/TileToScfForOp/TileOp/g"
> ```

@dcaballe is this entrenching the nan propagation issue even more?

**Comment by dcaballe** [comment]
That's part of what we need to integrate for the proper fix. The last one we need is https://github.com/llvm/llvm-project/pull/66431

**Comment by MaheshRavishankar** [comment]
> That's part of what we need to integrate for the proper fix. The last one we need is [llvm/llvm-project#66431](https://github.com/llvm/llvm-project/pull/66431)

Ok, I am assuming you are looking into this. Short of revert, I dont have much more context of what needs to be done here.

**Comment by dcaballe** [comment]
Ok, could you please add `createReplaceSlowMinMaxOpsPass` to the GPU pipeline only when <SM80? That should be it. It should be added when all the arith/vector max/min ops are around, before lowering them to LLVM.

File: compiler/src/iree/compiler/Codegen/Common/ReplaceSlowMinMaxOps.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The IREE Authors
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Common/ReplaceSlowMinMaxOps.cpp:1]: This kind of works too, but I was thinking go one level deeper and just do the rewrite at LLVM dialect level? 

@Groverkss ?

**Pull request merged**